### PR TITLE
Add new GamePad.GetState() overloads to support different dead zone modes

### PR DESCRIPTION
**Pull request by mrhelmut**
Hello !

This is a proposal for new public overloads to apply different ```GamePadDeadZone``` for each thumb sticks.

```public static GamePadState GetState(PlayerIndex playerIndex, GamePadDeadZone leftDeadZoneMode, GamePadDeadZone rightDeadZoneMode)```

```public static GamePadState GetState(int index, GamePadDeadZone leftDeadZoneMode, GamePadDeadZone rightDeadZoneMode)```

### CODE REVIEW

File: MonoGame.Framework/Input/GamePadThumbSticks.cs [hunk 1]
```diff
@@ -43,43 +43,62 @@ public Vector2 Right
...
                     break;
                 case GamePadDeadZone.Circular:
                     _left = ExcludeCircularDeadZone(leftPosition, leftThumbDeadZone);
-                    _right = ExcludeCircularDeadZone(rightPosition, rightThumbDeadZone);
                     break;
             }
 
             // Apply clamp
-            if (deadZoneMode == GamePadDeadZone.Circular)
+            if (rightDeadZoneMode == GamePadDeadZone.Circular)
```
[the remark this answers is not in the document]
Comment (mrhelmut) [reply MonoGame.Framework/Input/GamePadThumbSticks.cs]: @Jjagg Good call! I pushed a refactor!

**Comment by Jjagg** [comment]
Thanks @mrhelmut! Mergin'!

**Comment by HopefulToad** [comment]
Why didn't this PR add a separate overload instead of replacing the original overload that had just one deadzone type specified for both sticks? The way it is now, it's a breaking change. Both overloads could have existed, one where one deadzone type is specified for both sticks, and one where separate deadzone types are specified for each stick.

**Comment by mrhelmut** [comment]
You're probably referring to the internal/private methods, but the public API got extended, not replaced.

**Comment by HopefulToad** [comment]
Ah! Right you are, I didn't notice that the methods where the overload was being replaced were private, and that there was a public overload added. My bad.

**Pull request merged**